src/components/SiteTrackingRequired/SiteTrackingRequired.test.js

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import { render, cleanup, waitForDomChange } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import DopplerIntlProvider from '../../i18n/DopplerIntlProvider.double-with-ids-as-values';
+import { AppServicesProvider } from '../../services/pure-di';
+import { SiteTrackingRequired, SiteTrackingNotAvailableReasons } from './SiteTrackingRequired';
+
+describe('site tracking', () => {
+  afterEach(cleanup);
+
+  it('should show free account messages', () => {
+    // Arrange
+    const reason = SiteTrackingNotAvailableReasons.freeAccount;
+    const dependencies = {
+      dopplerLegacyClient: {
+        activateSiteTrackingTrial: async () => false,
+      },
+    };
+
+    // Act
+    const { getByText } = render(
+      <AppServicesProvider forcedServices={dependencies}>
+        <DopplerIntlProvider>
+          <SiteTrackingRequired reason={reason} />
+        </DopplerIntlProvider>
+      </AppServicesProvider>,
+    );
+
+    // Assert
+    expect(getByText('reports.upgrade_account_free_title')).toBeInTheDocument();
+  });
+
+  it('should show trial not accepted messages', () => {
+    // Arrange
+    const reason = SiteTrackingNotAvailableReasons.trialNotAccepted;
+    const dependencies = {
+      dopplerLegacyClient: {
+        activateSiteTrackingTrial: async () => false,
+      },
+    };
+
+    // Act
+    const { getByText } = render(
+      <AppServicesProvider forcedServices={dependencies}>
+        <DopplerIntlProvider>
+          <SiteTrackingRequired reason={reason} />
+        </DopplerIntlProvider>
+      </AppServicesProvider>,
+    );
+
+    // Assert
+    expect(getByText('reports.allow_enable_trial_title')).toBeInTheDocument();
+  });
+
+  it('should show not domains messages', () => {
+    // Arrange
+    const reason = SiteTrackingNotAvailableReasons.thereAreNotDomains;
+    const dependencies = {
+      dopplerLegacyClient: {
+        activateSiteTrackingTrial: async () => false,
+      },
+    };
+
+    // Act
+    const { getByText } = render(
+      <AppServicesProvider forcedServices={dependencies}>
+        <DopplerIntlProvider>
+          <SiteTrackingRequired reason={reason} />
+        </DopplerIntlProvider>
+      </AppServicesProvider>,
+    );
+
+    // Assert
+    expect(getByText('reports.datahub_not_domains_title')).toBeInTheDocument();
+  });
+
+  it('should be activate the trial and redirect to control panel site tracking settings', async () => {
+    // Arrange
+    const reason = SiteTrackingNotAvailableReasons.trialNotAccepted;
+    const dependencies = {
+      dopplerLegacyClient: {
+        activateSiteTrackingTrial: async () => true,
```

Comment:
It is not a successful result, why is the test passing?

The fix is easy:

```suggestion
        activateSiteTrackingTrial: async () => { success: true },
```

But, please try to identify why the test pass anyway.

Reply:
Also, what do you think about adding a test simulating failure in `activateSiteTrackingTrial` after the click?

Reply:
yes i done

Reply:
Tenés idea de por que no fallaba el test cuando simulabas una respuesta incorrecta?

```diff
- activateSiteTrackingTrial: async () => true,
+ activateSiteTrackingTrial: async () => ({ success: true }),
```

Reply:
porque estaba chequeando por la variable y no por el success en al componente

Reply:
Ahh, espectacular. Ya está corregido en el test y en el código entonces. Doy por resuelta  esta conversación. :)